dowhy/gcm/unit_change.py

```diff
@@ -0,0 +1,140 @@
+"""This module provides the APIs for attributing the change in the output value of a deterministic mechanism for a statistical unit.
+"""
+
+from abc import ABC, abstractmethod
+from typing import List, Optional
+
+import numpy as np
+import pandas as pd
+from sklearn.utils.validation import check_is_fitted
+from typing_extensions import Protocol
+
+from dowhy.gcm.shapley import ShapleyConfig, estimate_shapley_values
+
+
+class CanEvaluateInput(Protocol):
+    """This protocol defines a trait for classes that can evaluate a given input."""
+
+    @abstractmethod
+    def evaluate(self, input: np.ndarray) -> np.ndarray:
+        pass
```

Comment:
Instead of defining `CanEvaluateInput` explicitly, couldn't this just be `Mechanism` directly? `CanEvaluateInput` is not used anywhere. This would simplify the design a bit here.

Reply:
Fair point. But we might need it to make interfaces more consistent. See my comment below.